<?xml version="1.0" encoding="UTF-8"?>
<launch>
	
	<!-- Args -->
	<arg name="world_file"  default="$(find zed_slam)/worlds/sala_de_testes"/>

	<!-- Gazebo -->
	<include file="$(find gazebo_ros)/launch/empty_world.launch">
		<arg name="use_sim_time" 			value="true"/>
		<arg name="debug" 					value="false"/>
		<arg name="gui" 					value="true" />
		<arg name="world_name"				value="$(arg world_file)"/>
	</include>

	<!-- Spawn Roboserv-->
	<param name="robot_description" command="$(find xacro)/xacro.py '$(find roboserv_description)/urdf/roboserv.xacro'"/>
	<node name="roboserv_spawn" pkg="gazebo_ros" type="spawn_model" output="screen" args="-urdf -param robot_description -model roboserv" />
	
	<!-- send fake joint values -->
	<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
	<param name="use_gui" value="False"/>
	</node>

	<!-- Combine joint values -->
	<node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher"/>

	<!-- Show in Rviz   -->
	<node name="rviz" pkg="rviz" type="rviz"/>
	
	<!-- Launch pointcloud_to_laserscan_node -->
	<node name="pointcloud_to_laserscan_node" pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" output="screen" >
		<remap from="cloud_in" to="camera/depth/points"/>
		<param name="target_frame"		value="camera_link"/>
		<param name="range_min"         value="0.3"/>
		<param name="range_max"         value="15.0"/>
		<param name="min_height"        value="0.0"/>
		<param name="max_height"        value="0.6"/>
	</node>
	
	<!-- Velocity muxer -->
	<node pkg="nodelet" type="nodelet" name="mobile_base_nodelet_manager" args="manager"/>
	<node pkg="nodelet" type="nodelet" name="cmd_vel_mux"
		args="load yocs_cmd_vel_mux/CmdVelMuxNodelet mobile_base_nodelet_manager">
		<param name="yaml_cfg_file" value="$(find turtlebot_bringup)/param/mux.yaml" />
	</node>
	
	<!-- Launch gmapping -->
	<node name="gmapping" pkg="gmapping" type="slam_gmapping" output="screen" >
		<param name="base_frame" value="odom"/>
		<param name="agularUpdate" value="0.1"/>
		<param name="linearUpdate" value="0.1"/>
		<param name="lskip" value="10"/>
		<param name="xmax" value="10"/>
		<param name="xmin" value="-10"/>
		<param name="ymax" value="10"/>
		<param name="ymin" value="-10"/>
		<param name="minimumScore" value="50"/>
	</node>
	
	<!-- Move base -->
	<node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
		<rosparam file="$(find roboserv_simulation)/config/move_base/costmap_common_params.yaml" command="load" ns="global_costmap" />
		<rosparam file="$(find roboserv_simulation)/config/move_base/costmap_common_params.yaml" command="load" ns="local_costmap" />
		<rosparam file="$(find roboserv_simulation)/config/move_base/local_costmap_params.yaml" command="load" />
		<rosparam file="$(find roboserv_simulation)/config/move_base/global_costmap_params.yaml" command="load" />
		<rosparam file="$(find roboserv_simulation)/config/move_base/base_local_planner_params.yaml" command="load" />
		<param name="oscillation_timeout" value="0.5"/>
		
		<remap from="cmd_vel" to="cmd_vel_mux/input/navi"/>
	</node>
	
</launch>
